### PR TITLE
shell: shell_log_backend: take buffer size from config

### PR DESCRIPTION
**Pull request by wojciechslenska**
Use LOG_MESSAGE_QUEUE_SIZE config instead of hardcoded value. LOG_MESSAGE_QUEUE_SIZE default value has been changed to 512, so it is now matching to the hardcoded value.

Signed-off-by: Wojciech Slenska <wsl@trackunit.com>

### CODE REVIEW

**Review by mzagrabski — DISMISSED** [review]

**Review by nordic-krch — COMMENTED** [review]
Yeah, that was on my todo list for a long time. Looks good. Can you also edit template file `shell_log_queue_size` and rephrase it to mention buffer size in 32 bit words instead of  number of messages.

**Review by nordic-krch — APPROVED** [review]

**Comment by nordic-krch** [comment]
> Sure, but maybe better here will be in bytes, something like

My bad, it's actually in bytes.

**Review by stephanosio — APPROVED** [review]

**Comment by HrMitrev** [comment]
Hi!
I recently found out that this speciffic commit breaks my app(which is in Alpha stage). The problem is that the debug backtrace is broken after the crash and I am not sure right now if I was doing something wrong, which was masked before this commit, or this commit has a breaking change or bug.

```
***** USAGE FAULT *****
  Illegal load of EXC_RETURN into PC
r0/a1:  0xaa1671df  r1/a2:  0xf0284834  r2/a3:  0xe78cf8fb
r3/a4:  0x681b4b32 r12/ip:  0x0f07f013 r14/lr:  0xf04fd102
 xpsr:  0x93234a00
Faulting instruction address (r15/pc): 0xe7a035ff
>>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
Current thread: 0x20001c18 (£  )
Halting system
```

The app starts and runs for a few seconds before this crash. The crash seems to break the stack and I cannot easily debug without backtrace. If I revert the commit, the app works more than an hour(not tested extensively or for more than an hour).

Sorry for providing so little info, I'm just pointing that out and I don't have the time to open an issue with serious debug info.

**Comment by HrMitrev** [comment]
Hi!
I recently found out that this speciffic commit breaks my app(which is in Alpha stage). The problem is that the debug backtrace is broken after the crash and I am not sure right now if I was doing something wrong, which was masked before this commit, or this commit has a breaking change or bug.
```
***** USAGE FAULT *****
  Illegal load of EXC_RETURN into PC
r0/a1:  0xaa1671df  r1/a2:  0xf0284834  r2/a3:  0xe78cf8fb
r3/a4:  0x681b4b32 r12/ip:  0x0f07f013 r14/lr:  0xf04fd102
 xpsr:  0x93234a00
Faulting instruction address (r15/pc): 0xe7a035ff
>>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
Current thread: 0x20001c18 (£  �)
Halting system
```
The app starts and runs for a few seconds before this crash. The crash seems to break the stack and I cannot easily debug without backtrace. If I revert the commit, the app works more than an hour(not tested extensively or for more than an hour).

I only use the dummy shell backend(I only need to execute commands recieved via speciffic bluetooth service)

Sorry for providing so little info, I'm just pointing that out and I don't have the time to open an issue with serious debug info.
